### PR TITLE
configure.pl: Introduce --config-only option (#15)

### PR DESCRIPTION
**Pull request by moisseev**
This adds ability to skip installation phase and intended for using mostly
by packagers. It is assumed that the OS distribution package should install
configure.pl somewhere and provide a wrapper script for users like:

```sh
#!/bin/sh

PERL5LIB=/usr/local/lib

perl %%PREFIX%%/libexec/backuppc/configure.pl \
    --config-only
    --bin-path perl=%%PREFIX%%/bin/perl \
    --config-dir %%ETCDIR%% \
    --cgi-dir %%CGIDIR%% \
    --data-dir /var/db/BackupPC \
    --fhs \
    --html-dir %%WWWDIR%% \
    --html-dir-url /backuppc \
    --install-dir %%PREFIX%% \
    --log-dir /var/log/BackupPC
```

### CODE REVIEW

File: configure.pl [hunk 1]
```diff
@@ -514,165 +520,33 @@
...
-#
-# clean up old directory (ok if it quietly fails if there are other files in that directory)
-#
-rmdir("$DestDir$Conf{InstallDir}/doc") if ( -d "$DestDir$Conf{InstallDir}/doc" );
+DoInstall()
+  unless $opts{"config-only"};
 
 printf("Installing config.pl and hosts in $DestDir$Conf{ConfDir}\n");
-InstallFile("conf/hosts", "$DestDir$Conf{ConfDir}/hosts", 0644)
+InstallFile("conf/hosts", "$DestDir$Conf{ConfDir}/hosts.sample", 0644)
```
Comment (craigbarratt) [inline configure.pl]: I don't understand the creation of hosts.sample, which happens when !$opts{"config-only"}.  Why?

Comment (moisseev) [inline configure.pl]: This is all about making `configure.pl` more package friendly.

1. In general the `configure.pl` script is executed at package creation time and installs the program into a staging directory.
2. The preinstalled program files distributed as a package. A package installation is a basically copying that files into proper places on the hosts file sysytem.

3. After the package is installed user needs to run `configure.pl` again to upgrade `config.pl` from previous version (and maybe to configure new installation).
This time we should use `--config-only` option to skip installation phase.

Since `configure.pl` uses `conf/config.pl` and BackupPC source code or distribution is not available, we should add `conf/config.pl` to the package and install it somewhere.

It is a common good practice to provide sample configuration files that could be used by user as a reference or to fall back to default configuration.
Basically they are copies of configuration files installed by a package.

My idea is we can have `config.pl.sample` for both [3] and as a sample. That is why it gets created if !$opts{"config-only"} and used if !$opts{"config-only"}.

`hosts.sample` doesn't needed for `configure.pl`, it is just a sample file. So, it should be created only once at installation time.

I know it is not an elegant solution, but it doesn't involves major rewriting of the `configure.pl` and will substantially simplify creation of BackupPC packages. That is very complicated for now since `configure.pl` makes modifications at once in program and configuration files.

**Comment by craigbarratt** [comment]
Ok thanks for the explanation.  I merged this.  Thanks.
